mantidproject/mantid
# HB2C/HB3A workflow updates using mutiple goniometers

This modifies the algorithm used by HB2C and HB3A to make use of the multiple goniometers introduced in #30543.

The major changes being in the `ConvertHFIRSCDtoMDE` and `ConvertWANDSCDtoQ` algorithms where the goniometer is no longer _hand_ calculated but is instead pulled from the run object as set by `SetGoniometers` with `Average=False`. This greatly simplifies the algorithms.

**To test:**

Really just code review. There was already significant test coverage for the algorithms change and the tests have been updated with insignificant changes.

*This does not require release notes* because this is an internal change only and there should be no user-visible differences.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
